ambry-frontend/src/main/java/com.github.ambry.frontend/AmbrySecurityService.java

```diff
@@ -118,19 +118,19 @@ public AmbrySecurityService(FrontendConfig frontendConfig, FrontendMetrics front
             responseChannel.setStatus(ResponseStatus.Ok);
             RestUtils.SubResource subResource = RestUtils.getBlobSubResource(restRequest);
             if (subResource == null) {
+              responseChannel.setHeader(RestUtils.Headers.LAST_MODIFIED,
+                  new Date(blobInfo.getBlobProperties().getCreationTimeInMs()));
               Long ifModifiedSinceMs = getIfModifiedSinceMs(restRequest);
               if (ifModifiedSinceMs != null
                   && RestUtils.toSecondsPrecisionInMs(blobInfo.getBlobProperties().getCreationTimeInMs())
                   <= ifModifiedSinceMs) {
                 responseChannel.setStatus(ResponseStatus.NotModified);
-                responseChannel.setHeader(RestUtils.Headers.CONTENT_LENGTH, 0);
```

Comment:
I had code in `NettyResponseChannel` that set the transfer-encoding header to chunked if the content length was not set. So since there was no content, this content length was set to 0. According to the spec of Not Modified responses, a content-length field is prohibited - so this has to go.